docs/examples/single_screen.rst

```diff
@@ -0,0 +1,316 @@
+    :include-source:
+    :context:
+
+    plt.figure(figsize=(12., 3.))
+    plt.scatter(th_par, np.zeros_like(th_par),
+                s=np.abs(magnification)*2000., c=np.angle(magnification),
+                cmap=phasecmap, vmin=-np.pi, vmax=np.pi)
+
+    plt.xlabel(rf"$\theta_\parallel$ ({th_par.unit.to_string('latex')})")
+    plt.ylabel(rf"$\theta_\perp$ ({th_par.unit.to_string('latex')})")
+
+    cbar = plt.colorbar(aspect=7.5)
+    cbar.set_label('phase (rad)')
+    cbar.set_ticks([-np.pi, -np.pi/2., 0., np.pi/2., np.pi])
+    cbar.set_ticklabels([r'$-\pi$', r'$-\pi/2$', '0', r'$\pi/2$', r'$\pi$'])
+
+    plt.show()
+
+
+Set up observing parameters
+===========================
+
+Set the parameters that describe the observation:
+the central observing frequency :math:`f_\mathrm{obs}`,
+the bandpass :math:`\Delta f`,
+the observation length :math:`\Delta t`,
+the number of frequency channels :math:`n_f`,
+and the number of time bins :math:`n_t`.
+
+.. plot::
+    :include-source:
+    :context: close-figs
+
+    fobs = 316. * u.MHz
+    delta_f = 2. * u.MHz
+    delta_t = 90. * u.minute
+    nf = 200
+    nt = 180
+
+Set up a grid of observing frequencies and times.
+Then make the frequency grid a row vector and the time grid a column vector,
+so they will be broadcast against each other correctly.
+
+.. plot::
+    :include-source:
+    :context:
+
+    f = (fobs + np.linspace(-0.5*delta_f, 0.5*delta_f, nf, endpoint=False)
+         + 0.5*delta_f/nf)
+    t = np.linspace(0.*u.minute, delta_t, nt, endpoint=False) + 0.5*delta_t/nt
+
+    f, t = np.meshgrid(f, t, sparse=True)
+
+Already define an ``extent`` for plotting the electric field and dynamic
+spectrum.
+
+.. plot::
+    :include-source:
+    :context:
+
+    ds_extent = (t[0][0].value - 0.5*(t[1][0].value - t[0][0].value),
+                 t[-1][0].value + 0.5*(t[1][0].value - t[0][0].value),
+                 f[0][0].value - 0.5*(f[0][1].value - f[0][0].value),
+                 f[0][-1].value + 0.5*(f[0][1].value - f[0][0].value))
+
+
+Generate the electric field
+===========================
+
+Set the parameters of the system: the effective distance :math:`d_\mathrm{eff}`
+and the effective proper motion :math:`\mu_\mathrm{eff}`.
+
+.. plot::
+    :include-source:
+    :context:
+
+    d_eff = 0.5 * u.kpc
+    mu_eff = 50. * u.mas / u.yr
+
+Create electric fields for each of the scattering points, given by
+
+.. math::
+
+    E_i(f, t) = \mu_k \exp \left[ j f \frac{d_\mathrm{eff}}{2 c}
+                                  \theta_i^2 \right]
+
+.. plot::
+    :include-source:
+    :context:
+    
+    th_par_t = th_par[:,np.newaxis,np.newaxis] + mu_eff * t
+    theta_t_squared = th_par_t**2 + th_perp[:,np.newaxis,np.newaxis]**2
+    tau_t = (((d_eff / (2*const.c)) * theta_t_squared)
+             .to(u.s, equivalencies=u.dimensionless_angles()))
+
+    phasor = np.exp(1j * (f * tau_t * u.cycle).to_value(u.rad))
+    efields = phasor * magnification[:,np.newaxis,np.newaxis]
```

Comment:
Total nitpick, but add spaces after the commas.

Reply:
Done. I appreciate the nitpick!